File: Example/ViewController.m
//
//  ViewController.m
//  Example
//
//  Created by Mikhail Stepkin on 18.02.16.
//  Copyright © 2016 Ramotion. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
  [super viewDidLoad];
  // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
  [super didReceiveMemoryWarning];
  // Dispose of any resources that can be recreated.
}

- (void)viewDidLayoutSubviews {
  [super viewDidLayoutSubviews];

  [self.childViewControllers
      enumerateObjectsUsingBlock:^(__kindof UIViewController* _Nonnull obj,
                                   NSUInteger idx, BOOL* _Nonnull stop) {
        [obj viewDidLayoutSubviews];
      }];
}

@end

File: Example/Example/ViewController.swift
//
//  ViewController.swift
//  Example
//
//  Created by Mikhail Stepkin on 26.02.16.
//  Copyright © 2016 Ramotion. All rights reserved.
//

import UIKit
import MediaPicker

class ViewController: MediaSourceSwitcher {
    
    @IBOutlet var mediaPicker: MediaPicker!

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    override func viewDidLayoutSubviews() {
        super.viewDidLayoutSubviews()
        
        self.childViewControllers.forEach {
            $0.viewDidLayoutSubviews()
        }
    }

    override func viewDidAppear(animated: Bool) {
        super.viewDidAppear(animated)
        
        let switcher = MediaSourceSwitcher(maskType: .Circle, selectedColor: UIColor.redColor())
        self.presentViewController(switcher, animated: true, completion: nil)
    }

}

